File: application/src/main/java/whenyourcar/application/dto/chat/message/ChatMessageResponse.java
package whenyourcar.application.dto.chat.message;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import org.springframework.data.domain.Page;
import whenyourcar.domain.query.chat.SearchChatsQuery;

public class ChatMessageResponse {
    @Builder
    @Getter
    @AllArgsConstructor
    public static class ChatMessageResponseSetDto{
        private Long roomId;
        private String senderPicture;
        private String senderName;
        private Page<SearchChatsQuery> chatMessages;
    }

}

File: application/src/main/kotlin/howmuchcar/application/dto/chat/ChatMessageResponse.kt
package howmuchcar.application.dto.chat

import howmuchcar.domain.query.chat.SearchChatsQuery
import org.springframework.data.domain.Page

data class ChatMessageResponse (
    val roomId: Long,
    val senderPicture: String,
    val senderName: String,
    val chatMessages: Page<SearchChatsQuery>
){
}